frontend/packages/patternfly/src/components/notification-drawer/notification-entry.tsx

```diff
@@ -133,8 +155,11 @@ type NotificationIconTypes = {
 
 type NotificationActionProps = {
   onClick: (event: React.MouseEvent<HTMLElement>) => void;
-  text: string;
-  path: string;
+  text?: string;
+  path?: string;
+  alertAction?: Pick<ResolvedExtension<AlertAction>['properties'], 'text'> & {
+    actionProperty: () => void;
+  };
```

Comment:
It's more logical to let the parent component combine the alert action with `onClick`.